### ISSUE TITLE
validation errors in rawtherapee.appdata.xml

### ISSUE THREAD

**Issue by william8000** [Beep6581/RawTherapee#2954]
I am trying to make a CentOS 7 RPM. I started with a spec file from a recent Fedora.
The spec file validates the appdata file with
  `appstream-util validate-relax --nonet rawtherapee.appdata.xml`
The validation fails with
  `style-invalid         : Not enough <p> content before <ul>`
The patch below fixes the validation problem by including the `<ul>` inside the `<p>`.
Regards, William Bader, Director of Research and Development at SCS http://www.newspapersystems.com
```
--- rawtherapee-4.2/rawtherapee.appdata.xml-	2015-10-27 14:54:34.000000000 -0400
+++ rawtherapee-4.2/rawtherapee.appdata.xml	2015-10-27 20:21:25.892935832 -0400
@@ -15,7 +15,6 @@
         </p>
         <p>
             Supported input formats:
-        </p>
         <ul>
             <li>Most raw formats including those from Bayer and X-Trans type sensors</li>
             <li>HDR DNG 1.4 with floating-point data</li>
@@ -23,14 +22,15 @@
             <li>8-bit and 16-bit PNG (RGB)</li>
             <li>JPEG</li>
         </ul>
+        </p>
         <p>
             Supported output formats:
-        </p>
         <ul>
             <li>JPEG</li>
             <li>8-bit and 16-bit TIFF</li>
             <li>8-bit and 16-bit PNG</li>
         </ul>
+        </p>
     </description>
     <screenshots>
         <screenshot type="default">http://rawtherapee.com/images/screenshots/rt-42_07-hdr-landscape.jpg</screenshot>

```

**Comment by Beep6581** [comment]
https://github.com/Beep6581/RawTherapee/issues/2706#issuecomment-130064902
Including block-level elements inside <p> is forbidden in HTML. It validates in XML, but... I don't think catering to some validator's ridiculous and arbitrary requirements is good design. Their own examples from their homepage fail their own validator.
2>/dev/null

Their online validator doesn't work at all, and the offline appdata-validate shows:
 THIS TOOL IS *DEPRECATED* AND WILL BE REMOVED SOON.
 Please use 'apstream-util validate' in appstream-glib.

When I use that one (without the "apstream" typo in the name), your patch too fails validation.

I committed something that validates, but if there's any more trouble with this I'm inclined to just delete the damn thing.

**Comment by mbajor** [comment]
@william8000 It's strange. Validates succesfully on openSUSE with result:
rawtherapee.appdata.xml: OK

I'll check it for CentOS build on OBS.

I think the patch is invalid because putting lists in a paragraph element is forbidden: http://developers.whatwg.org/grouping-content.html#the-p-element

**Comment by mbajor** [comment]
@william8000 @Beep6581 Can't validate on OBS, because standard CentOS repositories don't provide appstream-util tool.

Latest version fails on Fedora 21 (libappstream-glib 0.2.7) and openSUSE (appstream-glib 0.2.6) but validates successfully on Fedora 22 (libappstream-glib 0.3.6) and rest of SUSE.
I think this tool contains bugs in older versions.

If you looking for CentOS 7 package, you can download it from our project on OBS:
https://software.opensuse.org/download.html?project=home%3Arawtherapee&package=rawtherapee

**Comment by william8000** [comment]
OK. Thanks for looking at it.
On CentOS 7, I built libappstream-glib-0.2.5 from source. The current 0.5.2 release had too many dependencies that were not available on CentOS 7.
`$ appstream-util --version`
shows
`Version:        0.2.5`
I looked at it more, and I agree with you that appstream-util is messed up and the original rawtherapee appdata file matches the example at http://people.freedesktop.org/~hughsient/appdata/

**Comment by william8000** [comment]
I missed some of the earlier replies. CentOS 7 doesn't have as many packages as Fedora, but when something is missing, I try to build an RPM from a recent Fedora source RPM. CentOS 7 is similar to Fedora 20. CentOS 7 does not have libappstream-glib, but I built my own RPM from a Fedora 20 (I think) source RPM. Usually I just build directly from source, but we are looking at rawtherapee and a few other packages, and I might need to install it on a number of servers. The RPM is more work to build but will save time. I have had problems with dependencies in RPMs from third-party repositories that block OS updates, so I prefer to build my own when it is easy.
I would also like to thank all of the developers for building a great application. I loaded a few photos and could optimize them without any problems.
Regards, William
